Control/Monad/Primitive.hs

```diff
@@ -51,51 +78,106 @@ primitive_ f = primitive (\s# ->
 instance PrimMonad IO where
   type PrimState IO = RealWorld
   primitive = IO
-  internal (IO p) = p
   {-# INLINE primitive #-}
+instance PrimMonadInternal IO where
+  internal (IO p) = p
```

Comment:
should this maybe be written as 
`internal = \(IO p)-> p` 
so that it has the same saturation/inlining behavior as other instances?